llvm/lib/Target/NVPTX/NVPTX.h

```diff
@@ -106,15 +106,49 @@ enum LoadStore {
   isStoreShift = 6
 };
 
-namespace PTXLdStInstCode {
-enum MemorySemantic {
+// Extends LLVM AtomicOrdering with PTX Orderings:
+using OrderingUnderlyingType = unsigned int;
+enum class Ordering : OrderingUnderlyingType {
   NotAtomic = 0, // PTX calls these: "Weak"
-  Volatile = 1,
+  // Unordered = 1, // NVPTX maps LLVM Unorderd to Relaxed
   Relaxed = 2,
-  Acquire = 3,
-  Release = 4,
-  RelaxedMMIO = 5
+  // Consume = 3,   // Unimplemented in LLVM; NVPTX would map to "Acquire"
+  Acquire = 4,
```

Comment:
I don't think we really really need them, but in the DAG they are just `(i32 x)` nodes where `x` is one of these values, and the values picked here match the ones of LLVM `AtomicOrdering` for the cases in which both expose the same ordering, which makes it easy to interpret those DAG nodes.

Reply:
Good idea with the `static_assert`s, that would prevent this from breaking.
Since we have some orderings that extend AtomicOrdering and we have to set `LAST`, etc., I've left the integer values though. If a `static_assert` fail, we'll need to update this properly.